### PR TITLE
Update patching guidelines

### PR DESCRIPTION
**Pull request by premun**

### CODE REVIEW

File: Documentation/patching-guidelines.md [hunk 1]
```diff
@@ -39,6 +39,8 @@ create one.
 > git format-patch --zero-commit --no-signature -1 -o "$PATCH_DIR/<repo>"
 > ```
 
+**Note:** Please make sure there are no other already existing patches which target same files as the newly added patch.
```
Comment (premun) [inline Documentation/patching-guidelines.md:42]:
```suggestion
**Note:** Please make sure there are no other already existing patches
which target same files as the newly added patch.
```

File: Documentation/patching-guidelines.md [hunk 2]
```diff
@@ -39,6 +39,9 @@ create one.
 > git format-patch --zero-commit --no-signature -1 -o "$PATCH_DIR/<repo>"
 > ```
 
+**Note:** Please make sure there are no other already existing patches
```
Comment (MichaelSimons) [inline Documentation/patching-guidelines.md:42]: Does this break synchronization?  It has been necessary in the past.

Comment (premun) [reply Documentation/patching-guidelines.md:42]: It breaks the synchronization, we've only had it once. Why would it be necessary?

Comment (MichaelSimons) [reply Documentation/patching-guidelines.md:42]: > Why would it be necessary?

Historically there have been numerous times where a single file is touched by multiple patches.  Patches are typically kept separate per the time they were made and their scope.  This makes patch removal much easier as the backports flow in.

Can't get backport flow soon enough...

Comment (premun) [reply Documentation/patching-guidelines.md:42]: In which order would the patches be applied? Alphabetically? I know that the one case I mention was one patch adding a file and another patching it

Comment (MichaelSimons) [reply Documentation/patching-guidelines.md:42]: alphabetically - that is why the numeric prefix is included in the guidelines.

Comment (premun) [reply Documentation/patching-guidelines.md:42]: https://github.com/dotnet/arcade-services/pull/4188

**Pull request closed without merging**